src/app/features/issue/providers/gitlab/gitlab-common-interfaces.service.ts

```diff
@@ -21,19 +23,15 @@ export class GitlabCommonInterfacesService implements IssueServiceInterface {
     private readonly _gitlabApiService: GitlabApiService,
     private readonly _projectService: ProjectService,
     private readonly _snackService: SnackService,
+    private readonly _issueCacheService: IssueCacheService,
   ) {
   }
 
   issueLink$(issueId: number, projectId: string): Observable<string> {
     return this._getCfgOnce$(projectId).pipe(
-      map((cfg) => {
-        if (cfg.gitlabBaseUrl != null) {
-          const fixedUrl = cfg.gitlabBaseUrl.match(/.*\/$/) ? cfg.gitlabBaseUrl : `${cfg.gitlabBaseUrl}/`;
-          return `${fixedUrl}${cfg.project}issues/${issueId}`;
-        } else {
-          return `${GITLAB_BASE_URL}${cfg.project?.replace(/%2F/g, '/')}/issues/${issueId}`;
-        }
-      })
+      flatMap((cfg) =>
```

Comment:
I would prefer not to make this change, as even with caching in place, these are requests which are not needed most of the time. It should be possible to derive the url from the id somehow.

Reply:
I tried many ways with the project ID without any luck. I have another idea, I'll implement as soon as possible (be patient ;). Another question, how can I add a hook to the GitLab/GitHub save method to remove cached variables (example: access token changed) ?

Reply:
new solution pushed: now I fetch the project (cached, 1 day expire) and get the "prefix" from it
ps: maybe we can create a "clear cache" button to clear cached variables of the project